README.md

```diff
@@ -1,116 +1,267 @@
-This is a readme for the RETNET team
+# Are You Using Retentive Networks? (RetNets) 📝
 
-Make sure to have Python=3.11
-When you use a .sh script, run this to grant permission "chmod +x retnet.sh" for the script you want to use
-Then do this "./retnet.sh" to run the script
+This project focuses on the comparison and analysis of the RetNet vs the Transformer architecture, utilizing Microsoft's TorchScale library for implementation. More information can be found in our paper [Are You Using Retentive Networks?](./301R_Retnet_Paper.pdf).
 
-_Hyperparameters Scaling_
+## Reference to Original Paper
 
-### Python Code Examples
+Our study is based on research detailed in the paper [Retentive Network: A Successor to Transformer for Large Language Models](https://arxiv.org/abs/2307.08621). For more in-depth information and methodology of the RetNet architecture, refer to this paper.
 
-#### 0. **Preparing Data**:
-This repository is designed to be run on a cluster. As such, the data must be downloaded and prepared before running the training script.
+## Base Architecture - Microsoft TorchScale
 
-To download data from the HuggingFace Hub, run the `download_data.py` script with the following parameters:
-- dataset_name : the name of the dataset on HF hub
-- dataset_subset : the subset to use
-- dataset_dir : the name of the folder to save the dataset in. Typically, you can just keep this the same as the dataset_name
+This project is built upon [Microsoft TorchScale](https://github.com/microsoft/torchscale), which provides basic implementations of the two architectures for our research. TorchScale provides a library of foundational architecture implementations for training Transformer-based deep learning models. We have leveraged its capabilities to develop our comparison between RetNets and Transformers.
 
-#### 1. **Setting Up a Basic Model**:
+## Installation and Setup
 
-This snippet shows how to initialize a neural network with basic hyperparameters.
+To get started with this project, first clone this repository using the following command:
 
-```python
-import torch
-import torch.nn as nn
-from your_project_model import RetNetModel  # Replace with your actual model import
+```bash
+git clone https://github.com/DRAGNLabs/301r_retnet.git
+cd 301r_retnet
+```
 
-# Basic hyperparameters
-hyperparams = {
-    "learning_rate": 0.001,
-    "batch_size": 64,
-    "num_layers": 4,
-    "embed_dim": 256,
-    "num_heads": 8,
-    "dropout_rate": 0.1
-}
+### Python Version Requirement
 
-model = RetNetModel(num_layers=hyperparams["num_layers"],
-                    embed_dim=hyperparams["embed_dim"],
-                    num_heads=hyperparams["num_heads"],
-                    dropout_rate=hyperparams["dropout_rate"])
-optimizer = torch.optim.Adam(model.parameters(), lr=hyperparams["learning_rate"])
+Ensure you have Python 3.11 installed. To check your current Python version, run:
+
+```bash
+python --version
```

Comment:
I don't know. I just thought that know what version of Python to use is important since some features will not work in other versions. @JayOrten @nprisbrey

Reply:
I find value in specifying what version of Python we're using, though it may be extra and unusual. I think that anything additional, such as explaining how to check the Python version, is equivalent to treating the reader as a child. Documentation and Google exist to help with this.

Reply:
I deleted that instruction.